#VRML_SIM R2022b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/floors/protos/RectangleArena.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/robots/picaxe/microbot/protos/Microbot.proto"

WorldInfo {
}
Viewpoint {
  orientation -0.599462885145109 0.6002124844464235 0.5295179155119815 2.166599098867877
  position -0.12523265323289068 0.7192467194164589 8.545264094388868
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  translation -0.06 -0.25 0
  floorSize 4.9 3
  floorAppearance PBRAppearance {
    roughness 1
    roughnessMap ImageTexture {
      url [
        "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/appearances/protos/textures/parquetry/chequered_parquetry_roughness.jpg"
      ]
    }
    metalness 0
    normalMap ImageTexture {
      url [
        "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/appearances/protos/textures/parquetry/chequered_parquetry_normal.jpg"
      ]
    }
    occlusionMap ImageTexture {
      url [
        "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/appearances/protos/textures/parquetry/chequered_parquetry_occlusion.jpg"
      ]
    }
  }
}
Solid {
  translation 2.01 -1.05 -7.55587e-06
  rotation -1 0 0 -5.307179586466759e-06
  scale 0.019 0.019 0.019
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 0 0
      }
      geometry Mesh {
        url [
          "C:/Users/fahri/Downloads/sulawesi/tinker.obj"
        ]
      }
    }
  ]
  name "sulawesi"
}
Solid {
  translation -1.74191 -0.114483 -2.59096e-06
  rotation -2.0155999995834707e-05 -2.653589999451628e-06 0.9999999997933471 -0.261795307179586
  scale 0.009 0.009 0.009
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 0 0
      }
      geometry Mesh {
        url [
          "C:/Users/fahri/Downloads/arena.s/tinker.obj"
        ]
      }
    }
  ]
  name "sumatera"
}
Solid {
  translation 0.432226 -1.27254 -8.73673e-06
  rotation 2.015529999583514e-05 -2.6534999994516847e-06 0.9999999997933615 0.261805
  scale 0.02 0.02 0.02
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 0 0
      }
      geometry Mesh {
        url [
          "C:/Users/fahri/Downloads/DesainArena/tinker.obj"
        ]
      }
    }
  ]
  name "jawa"
}
Solid {
  translation -0.22908 -0.466307 -4.45775e-06
  rotation 2.015529999583514e-05 -2.6534999994516847e-06 0.9999999997933615 0.261805
  scale 0.01 0.01 0.01
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 0 0
      }
      geometry Mesh {
        url [
          "C:/Users/fahri/Downloads/KaliMANTAN/tinker.obj"
        ]
      }
    }
  ]
  name "kalimantan"
}
Microbot {
  hidden position_0_0 -1480.658947356482
  hidden position_0_1 -449.4774945228894
  hidden translation_11 5.7568130940932605e-05 3.979809349630159e-05 1.6333955030201075e-07
  hidden rotation_11 -0.22241267794821967 -0.6966823449866447 -0.6820310189953322 2.6070066112878667e-15
  hidden translation_12 6.26591722301395e-05 -3.118999964557556e-05 1.2377332982843603e-07
  hidden rotation_12 -0.05580251426552809 0.44707404983827387 -0.8927546546296188 1.989557351022434e-15
  hidden rotation_14 0 -0.9999999999999999 0 2.1727851379003775
  hidden rotation_15 0 -1 0 2.9118475940408253
  translation 0.973611 0.503433 0.0271208
  rotation 0.0278282040206339 0.001073210155057981 0.9996121444245007 0.122755
  controller "universal_fuzzy_line_follower"
  extensionSlot [
    DistanceSensor {
      translation 0.045 -0.05 8.98039e-08
      rotation 0 -1 0 -1.5708053071795867
      name "ir0"
      type "infra-red"
    }
    DistanceSensor {
      translation 0.055 -0.04 0
      rotation 0 -1 0 -1.5708053071795867
      name "ir1"
      type "infra-red"
    }
    DistanceSensor {
      translation 0.057 -0.01 0
      rotation 0 -1 0 -1.5708053071795867
      name "ir2"
      type "infra-red"
    }
    DistanceSensor {
      translation 0.057 0 0
      rotation 0 -1 0 -1.5708053071795867
      name "ir3"
      type "infra-red"
    }
    DistanceSensor {
      translation 0.057 0.01 0
      rotation 0 -1 0 -1.5708053071795867
      name "ir4"
      type "infra-red"
    }
    DistanceSensor {
      translation 0.055 0.04 0
      rotation 0 -1 0 -1.5708053071795867
      name "ir5"
      type "infra-red"
    }
    DistanceSensor {
      translation 0.045 0.05 8.98038e-08
      rotation 0 -1 0 -1.5708053071795867
      name "ir6"
      type "infra-red"
    }
  ]
}
